src/matmul.jl

```diff
@@ -17,85 +16,59 @@ const SparseMatrices{T} = Union{SparseMatrixCSC{T,BlasInt},
             :($w{$T,SparseMatrixCSC{$T,BlasInt}}) :
             :($ttype{$T,$w{$T,SparseMatrixCSC{$T,BlasInt}}})
+        BT = :(Union{StridedMatrix{$T}, StridedVector{$T}})
+
+        # mul!(C, A, B, a, b), where A is Symmetric/LowTri etc of a SparseMatrixCSC or adjoint/transpose of it
+        # it has special implementation in Base, so we redefine it to use MKLSparse implementation
         @eval begin
-            function mul!(C::$mat{$T}, adjA::$AT, B::$mat{$T}, α::Number, β::Number)
-                A = _unwrap_adj(adjA)
-                if isa(B,AbstractVector)
-                    return cscmv!($tchar, $T(α), matdescra(A), _get_data(A), B, $T(β), C)
-                else
-                    return cscmm!($tchar, $T(α), matdescra(A), _get_data(A), B, $T(β), C)
-                end
-            end
+            LinearAlgebra.mul!(C::StridedVector{$T}, A::$AT, B::StridedVector{$T}, α::Number, β::Number) =
+                cscmv!($tchar, $T(α), describe_and_unwrap(A)..., B, $T(β), C)
 
-            mul!(C::$mat{$T}, adjA::$AT, B::$mat{$T}) = mul!(C, adjA, B, one($T), zero($T))
+            LinearAlgebra.mul!(C::StridedMatrix{$T}, A::$AT, B::StridedMatrix{$T}, α::Number, β::Number) =
+                cscmm!($tchar, $T(α), describe_and_unwrap(A)..., B, $T(β), C)
 
-            function (*)(adjA::$AT, B::$mat{$T})
-                A = _unwrap_adj(adjA)
-                if isa(B,AbstractVector)
-                    return mul!(zeros($T, mkl_size($tchar, A)[1]),            adjA, B)
-                else
-                    return mul!(zeros($T, mkl_size($tchar, A)[1], size(B,2)), adjA, B)
-                end
-            end
+            LinearAlgebra.mul!(C::BT, A::$AT, B::BT) where BT <: $BT = mul!(C, A, B, one($T), zero($T))
+
+            # base A*B converts A to dense, so we redefine it to use MKLSparse-enabled mul!
+            Base.:(*)(A::$AT, B::StridedVector{$T}) = mul!(Vector{$T}(undef, size(A, 1)), A, B)
+            Base.:(*)(A::$AT, B::StridedMatrix{$T}) = mul!(Matrix{$T}(undef, size(A, 1), size(B, 2)), A, B)
         end
 
+        # define 4-arg ldiv!(C, A, B, a) (C := alpha*inv(A)*B) that is not present in standard LinearAlgrebra,
+        # redefine 3-arg ldiv!(C, A, B) using 4-arg ldiv!(C, A, B, 1)
+        # here A is LowerTri/UpperTri etc of a SparseMatrixCSC or adjoint/transpose of it (Symmetric not supported)
         if w != :Symmetric
             @eval begin
-                function ldiv!(α::Number, adjA::$AT,
-                               B::$mat{$T}, C::$mat{$T})
-                    A = _unwrap_adj(adjA)
-                    if isa(B,AbstractVector)
-                        return cscsv!($tchar, $T(α), matdescra(A), _get_data(A), B, C)
-                    else
-                        return cscsm!($tchar, $T(α), matdescra(A), _get_data(A), B, C)
-                    end
-                end
+                LinearAlgebra.ldiv!(C::StridedVector{$T}, A::$AT, B::StridedVector{$T}, α::Number) =
```

Comment:
Could we document this? It matches `mul!` so it's not entirely unexpected

Reply:
You mean add the docstring?